catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -66,147 +131,867 @@ public FederationAdminServiceImpl() {
     }
 
-    public void setRegistryTransformer(InputTransformer inputTransformer) {
-        this.registryTransformer = inputTransformer;
+    @Override
+    public String addRegistryEntry(Metacard metacard, Set<String> destinations)
+            throws FederationAdminException {
+        validateRegistryMetacard(metacard);
+
+        String metacardId;
+
+        Subject systemSubject = security.getSystemSubject();
+        Map<String, Serializable> properties = new HashMap<>();
+        properties.put(SecurityConstants.SECURITY_SUBJECT, systemSubject);
+        CreateRequest createRequest = new CreateRequestImpl(Collections.singletonList(metacard),
+                properties,
+                destinations);
+
+        try {
+            CreateResponse response = catalogFramework.create(createRequest);
+            metacardId = response.getCreatedMetacards()
+                    .get(0)
+                    .getId();
+        } catch (IngestException | SourceUnavailableException e) {
+            throw new FederationAdminException("Error adding local registry entry.", e);
+        }
+
+        return metacardId;
     }
 
     @Override
-    public String addRegistryEntry(String xml) throws FederationAdminException {
-        return null;
+    public void updateRegistryEntry(String xml) throws FederationAdminException {
+        Metacard updateMetacard = getRegistryMetacardFromString(xml);
+        updateRegistryEntry(updateMetacard);
     }
 
     @Override
-    public String addRegistryEntry(String xml, Set<String> destinations)
+    public void updateRegistryEntry(String xml, Set<String> destinations)
             throws FederationAdminException {
-        return null;
+        Metacard updateMetacard = getRegistryMetacardFromString(xml);
+        updateRegistryEntry(updateMetacard, destinations);
     }
 
     @Override
-    public String addRegistryEntry(Metacard metacard) throws FederationAdminException {
-        return null;
+    public void updateRegistryEntry(Metacard updateMetacard) throws FederationAdminException {
+        updateRegistryEntry(updateMetacard, null);
     }
 
     @Override
-    public String addRegistryEntry(Metacard metacard, Set<String> destinations)
+    public void updateRegistryEntry(Metacard updateMetacard, Set<String> destinations)
             throws FederationAdminException {
-        return null;
+        validateRegistryMetacard(updateMetacard);
+
+        if (updateMetacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID) == null) {
+            throw new FederationAdminException(
+                    "Error updating registry entry. Metacard registry-id is not set.");
+        }
+        String registryId = updateMetacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
+                .toString();
+        List<Filter> filters = new ArrayList<>();
+        filters.add(FILTER_FACTORY.like(FILTER_FACTORY.property(RegistryObjectMetacardType.REGISTRY_ID),
```

Comment:
Any reason why you are not injecting a filter builder?  In general we use the filter builder instead of directly using the filter factory.